### PR TITLE
MULE-13639 Remove FlowConstruct and MuleContext from Event/Message

### PR DESCRIPTION
**Pull request by dfeist**

### CODE REVIEW

File: core-tests/src/test/java/org/mule/runtime/core/api/event/MuleEventTestCase.java [hunk 1]
```diff
@@ -131,11 +125,15 @@ public void testMuleEventSerializationWithRawPayload() throws Exception {
...
-
-    assertArrayEquals((byte[]) testEvent.getMessage().getPayload().getValue(), payload.toString().getBytes());
+    try {
+      currentMuleContext.set(muleContext);
+      byte[] serializedEvent = muleContext.getObjectSerializer().getExternalProtocol().serialize(testEvent);
+      testEvent = muleContext.getObjectSerializer().getExternalProtocol().deserialize(serializedEvent);
+
+      assertArrayEquals((byte[]) testEvent.getMessage().getPayload().getValue(), payload.toString().getBytes());
+    } finally {
+      currentMuleContext.set(null);
```
Comment (elrodro83) [inline core-tests/src/test/java/org/mule/runtime/core/api/event/MuleEventTestCase.java]: use an `@After` method

File: core-tests/src/test/java/org/mule/runtime/core/internal/message/DefaultMuleMessageSerializationTestCase.java [hunk 2]
```diff
@@ -47,23 +48,31 @@ public void testNonSerializablePayload() throws Exception {
...
+    try {
+      currentMuleContext.set(muleContext);
+      InternalMessage deserializedMessage = serializationRoundtrip(message);
 
-    assertTrue(deserializedMessage.getPayload().getValue() instanceof byte[]);
-    assertEquals(INNER_TEST_MESSAGE, getPayloadAsString(deserializedMessage));
+      assertTrue(deserializedMessage.getPayload().getValue() instanceof byte[]);
+      assertEquals(INNER_TEST_MESSAGE, getPayloadAsString(deserializedMessage));
+    } finally {
+      currentMuleContext.set(null);
```
Comment (elrodro83) [inline core-tests/src/test/java/org/mule/runtime/core/internal/message/DefaultMuleMessageSerializationTestCase.java]: `@After`

File: core-tests/src/test/java/org/mule/runtime/core/internal/message/DefaultMuleMessageSerializationTestCase.java [hunk 3]
```diff
@@ -47,23 +48,31 @@ public void testNonSerializablePayload() throws Exception {
...
-    byte[] payload = (byte[]) deserializedMessage.getPayload().getValue();
-    assertTrue(Arrays.equals(TEST_MESSAGE.getBytes(), payload));
+    try {
+      currentMuleContext.set(muleContext);
+      InternalMessage deserializedMessage = serializationRoundtrip(message);
+      assertEquals(byte[].class, deserializedMessage.getPayload().getDataType().getType());
+      byte[] payload = (byte[]) deserializedMessage.getPayload().getValue();
+      assertTrue(Arrays.equals(TEST_MESSAGE.getBytes(), payload));
+    } finally {
+      currentMuleContext.set(null);
```
Comment (elrodro83) [inline core-tests/src/test/java/org/mule/runtime/core/internal/message/DefaultMuleMessageSerializationTestCase.java]: `@After`

File: core/src/main/java/org/mule/runtime/core/internal/message/DefaultMessageBuilder.java [hunk 4]
```diff
@@ -554,11 +559,15 @@ protected void serializeValue(ObjectOutputStream out) throws Exception {
       } else {
         out.writeBoolean(false);
         // TODO MULE-10013 remove this logic from here
```
Comment (elrodro83) [inline core/src/main/java/org/mule/runtime/core/internal/message/DefaultMessageBuilder.java]: update references to this issue

Comment (dfeist) [reply core/src/main/java/org/mule/runtime/core/internal/message/DefaultMessageBuilder.java]: This reference is still correct isn't it?

**Pull request merged**